File: front-pokedex-pokemon-card/src/app/Models/pokemonCard.ts
import { CardPrinting } from "./cardPrinting";
import { Sets } from "./sets";

export interface PokemonCard {
    id: number;
    localId: number;
    extension: string;
    name: string;
    image: string;
    pokemonId: number;
    set: Sets;
    cardPrintings: CardPrinting[];
}


export enum PokemonCardTypeSelected {
    Owned,
    Wanted
}

export interface InjectPokemonCardData {
    cards: PokemonCard[];
    type: PokemonCardTypeSelected;
    pokedexId: number;
    pokemonId: number;
}

File: API-pokedex-pokemon-card/Models/PokemonCard.cs
using System.ComponentModel.DataAnnotations.Schema;
using API_pokedex_pokemon_card.Models;

public class PokemonCard
{
    public required string Id { get; set; }
    public required string LocalId { get; set; }
    public required string Extension { get; set; }
    public required string Name { get; set; }
    public required string Image { get; set; }
    public List<PokemonCardPokemon> PokemonCardPokemons { get; set; } = new();
    public required Sets Set { get; set; }
    public int SetId { get; set; }
    
    public ICollection<CardPrinting> CardPrintings { get; set; } = new List<CardPrinting>();

    [NotMapped]
    public int PokemonId { get; set; }

}

public enum PokemonCardTypeSelected
{
    Owned,
    Wanted
}